Scripts/GetAutoShutdownPolicyStatus/GetAutoShutdownPolicyStatus.ps1

```diff
@@ -0,0 +1,98 @@
+    [String] $LabName
+
+)
+
+$console = (Get-Host).PrivateData
+$console.VerboseForegroundColor = "Gray"
+$console.WarningForegroundColor = "Yellow"
+
+if ($SubscriptionIds -eq $null) {
+    $SubscriptionIds = Get-AzureRmSubscription
+}
+
+foreach ($subscription in $SubscriptionIds) {
+
+    # select the subscription
+    $sub = Select-AzureRmSubscription -SubscriptionId $subscription
+
+    if ($sub -eq $null) {
+        Write-Output "Unable to find any subscriptions.  Perhaps you need to run 'Add-AzureRmAccount' and login before running this script? Unable to proceed."
+        return
+    }
+
+    # Give me all labs in the subscription
+    if ($LabName) {
+        Write-Verbose "Getting lab $($LabName)"
+        $devTestLabs = Find-AzureRmResource -ResourceType 'Microsoft.DevTestLab/labs' -ResourceNameContains $LabName
+    }
+    else {
+        $devTestLabs = Find-AzureRmResource -ResourceType 'Microsoft.DevTestLab/labs'
+    }
+    
+    foreach ($devTestLab in $devTestLabs) {
+        Write-Output "Processing lab $($devTestLab.Name)..."
+
+        $schedule = $null
+        $labHealthy = $true
+
+        # Get the AS policy status in the lab
+        try {
+            $schedule = Get-AzureRmResource -ResourceId "$($devTestLab.ResourceId)/schedules/LabVmsShutdown" -ApiVersion 2016-05-15
+        }
+        catch {
+            Write-Warning "Lab $($devTestLab.Name) auto-shutdown policy is disabled or no longer present"
+            $labHealthy = $false
+            continue
+        }
+
+        $schedDisabled = $false
+
+        if (($schedule -eq $null) -or ($schedule.Properties.status -ne 'Enabled')) {
+            Write-Warning "Lab $($devTestLab.Name) auto-shutdown policy is disabled"
+            $schedDisabled = $true
+            $labHealthy = $false
+        }
+
+        if (-not $schedDisabled) {
+            # Get all the VMs in the lab for an override
+            $virtualMachines = Get-AzureRmResource -ResourceId "$($devTestLab.ResourceId)/virtualmachines" -ApiVersion 2016-05-15 
+
+            foreach ($virtualMachine in $virtualMachines) {
+                Write-Verbose "   >>> Processing virtual machine $($virtualMachine.Name)"
+
+                # Get the tag and use it to determine AS status
+                $tags = $virtualMachine.Tags;
+
+                if ($tags -and $tags["AlwaysOn"] -eq "true") {
+                    Write-Warning "   >>>> Virtual machine $($virtualMachine.Name) auto-shutdown status is disabled"
+                    $labHealthy = $false
+                }
+
+                $vmSchedule = $null
+
+                try {
+                    $vmSchedule = Get-AzureRmResource -ResourceId "$($devTestLab.ResourceId)/virtualmachines/$($virtualMachine.Name)/schedules/LabVmsShutdown" -ApiVersion 2016-05-15
+                }
+                catch {
+                        
+                }
+    
+                if (($vmSchedule -ne $null) -and ($vmSchedule.Properties.status -ne 'Enabled')) {
```

Comment:
Collapse condition.
```powershell
if ($vmSchedule.Properties.status -ne 'Enabled')
```

Reply:
I need the null check here because it was a precondition that I check afterwards.  Removing the null means I will have to change the way I work with the 'no schedule found' scenario.   Leaving this one as is for now.